File: src/DisplayPanel.java
import javax.swing.*;
import java.awt.*;
import java.awt.image.BufferedImage;

public class DisplayPanel extends JPanel {

    private Image tempImage;

    private final Memory memory;
    private final PPU ppu;
    private final DisplayFrame displayFrame;

    private final int BACK_WINDOW_PALLETE = 0xff47;

    private final int WIDTH;
    private final int HEIGHT;

    //Constructor

    public DisplayPanel(Memory memory, PPU ppu, DisplayFrame displayFrame) {
        this.memory = memory;
        this.ppu = ppu;
        this.displayFrame = displayFrame;
        ppu.setDisplayPanel(this);

        WIDTH = displayFrame.getWidth();
        HEIGHT = displayFrame.getHeight();
    }

    public void drawImage(byte[][] painting) {
        BufferedImage bufferedImage = new BufferedImage(WIDTH, HEIGHT, BufferedImage.TYPE_INT_RGB);

        int[] size = displayFrame.getCurrentSize();
        this.setSize(new Dimension(size[0], size[1]));
        displayFrame.setHeight(size[1] + 28);
        displayFrame.setWidth(size[0]);

        Graphics g = bufferedImage.getGraphics();

        for(int x = 0; x < 160; x++) {
            for(int y = 0; y < 144; y++) {
                Color c = getColor(painting[x][y]);
                g.setColor(c);
                g.fillRect(x, y, 1, 1);
            }
        }

        tempImage = bufferedImage;
    }

    public void drawBlankImage() {
        BufferedImage bufferedImage = new BufferedImage(WIDTH, HEIGHT, BufferedImage.TYPE_INT_RGB);

        int[] size = displayFrame.getCurrentSize();
        this.setSize(new Dimension(size[0], size[1]));
        displayFrame.setHeight(size[1] + 28);
        displayFrame.setWidth(size[0]);

        Graphics g = bufferedImage.getGraphics();

        g.setColor(Color.WHITE);
        g.fillRect(0, 0, 160, 144);

        tempImage = bufferedImage;
    }


    public void paint(Graphics g) {
        int[] size = displayFrame.getCurrentSize();
        if(tempImage != null) {
            Image newImage = tempImage.getScaledInstance(size[0], size[1], Image.SCALE_DEFAULT);
            g.drawImage(newImage, 0, 0, this);
        }
    }

    private Color getColor(byte pixelNumber) {
        return switch (pixelNumber) {
            case 0 -> new Color(0xffffff);
            case 1 -> new Color(0xcccccc);
            case 2 -> new Color(0x777777);
            case 3 -> new Color(0x000000);
            default -> Color.RED;
        };
    }
}

File: kboyemucore/src/main/java/com/github/rodrigotimoteo/kboyemucore/DisplayPanel.kt
package com.github.rodrigotimoteo.kboyemucore

import com.github.rodrigotimoteo.kboyemucore.memory.Memory
import com.github.rodrigotimoteo.kboyemucore.ppu.PPU

class DisplayPanel {
//
//    private var tempImage: android.media.Image? = null
//
//    private val memory: Memory? = null
//    private val ppu: PPU? = null
//    private val displayFrame: com.github.rodrigotimoteo.kboyemu.kotlin.DisplayFrame? = null
//
//    private const val BACK_WINDOW_PALLETE = 0xff47
//
//    private const val WIDTH = 0
//    private const val HEIGHT = 0
//
//    private var cgb = false
//
//
//    //Constructor
//    fun DisplayPanel(
//        memory: Memory?,
//        ppu: PPU,
//        displayFrame: DisplayFrame
//    ) {
//        this.memory = memory
//        this.ppu = ppu
//        this.displayFrame = displayFrame
//        ppu.setDisplayPanel(this)
//
//        WIDTH = displayFrame.getWidth()
//        HEIGHT = displayFrame.getHeight()
//    }
//
//    fun setCgbMode() {
//        cgb = true
//    }
//
//    fun drawImage(painting: Array<ByteArray?>) {
//        val bufferedImage: BufferedImage = BufferedImage(WIDTH, HEIGHT, BufferedImage.TYPE_INT_RGB)
//
//        val size: IntArray = displayFrame.getCurrentSize()
//        this.setSize(Dimension(size[0], size[1]))
//        displayFrame.setHeight(size[1] + 28)
//        displayFrame.setWidth(size[0])
//
//        val g: Graphics = bufferedImage.getGraphics()
//
//        for (x in 0..159) {
//            for (y in 0..143) {
//                val c: androidx.compose.ui.graphics.Color? = getColor(painting[x]!![y])
//                g.setColor(c)
//                g.fillRect(x, y, 1, 1)
//            }
//        }
//
//        tempImage = bufferedImage
//    }
//
//    fun drawBlankImage() {
//        val bufferedImage: BufferedImage = BufferedImage(WIDTH, HEIGHT, BufferedImage.TYPE_INT_RGB)
//
//        val size: IntArray = displayFrame.getCurrentSize()
//        this.setSize(Dimension(size[0], size[1]))
//        displayFrame.setHeight(size[1] + 28)
//        displayFrame.setWidth(size[0])
//
//        val g: Graphics = bufferedImage.getGraphics()
//
//        g.setColor(WHITE)
//        g.fillRect(0, 0, 160, 144)
//
//        tempImage = bufferedImage
//    }
//
//
//    fun paint(g: Graphics) {
//        val size: IntArray = displayFrame.getCurrentSize()
//        if (tempImage != null) {
//            val newImage: android.media.Image? =
//                tempImage.getScaledInstance(size[0], size[1], android.media.Image.SCALE_DEFAULT)
//            g.drawImage(newImage, 0, 0, this)
//        }
//    }
//
//    private fun getColor(pixelNumber: Byte): androidx.compose.ui.graphics.Color? {
//        if (cgb) {
//        } else {
//            return when (pixelNumber) {
//                0 -> Color(0xffffff)
//                1 -> Color(0xcccccc)
//                2 -> Color(0x777777)
//                3 -> Color(0x000000)
//                else -> RED
//            }
//        }
//        return null
//    }
}